docs: add CDK and Template 1.0 announcement (#1510)
rx-angular/rx-angular: apps/docs/blog/2023-02-04-announcing-rxangular-cdk-and-template-1-0.md

---
title: Announcing the RxAngular CDK and Template 1.0 releases
description: The @rx-angular/cdk and @rx-angular/template packages are now stable and follow semantic versioning.
authors:
  - BioPhoton
  - edbzn
  - hoebbelsB
  - Karnaukhov-kh
  - LayZeeDK
tags: [cdk, release, template]
image: /img/logo.png
hide_table_of_contents: false
---

- [ ] Supported Angular versions
- [x] We follow Semantic Versioning
- [ ] Recent breaking changes/deprecations?
- [x] Introduction to each package
- [ ] Recent features in each package
- [x] Major features in each package

We are pleased to announce the stable release of RxAngular CDK and RxAngular Template 1.0. With this release, we enter semantic versioning which means no breaking changes in minor or patch version releases. This includes version requirements for our peer dependencies, Angular and RxJS.

Thank you to all contributors and users who have helped us to get here. We are excited to see what the future holds for RxAngular.

## Packages introduction

### @rx-angular/template

> The @rx-angular/template library helps simplify building Angular applications of any scale.
> It reduces the need for boilerplate code and provides fast and reliable rendering.
> It has two options for implementation - with and without zone.js - both optimized for performance.

Reviewer: ```suggestion
The @rx-angular/template library simplifies building Angular applications of any scale.
It improves the DX by reducing boilerplate code and provides fast and reliable change detection.
It's implementation is zone agnostic by default and optimizes internally regardless of the environment.
By leveraging the `@rx-angular/cdk/render-strategies` package, it enables the concurrent mode for #angular
```

Our CDK provides zone-configuration utils for partially disabling zone.js, reducing the risk of unexpected bugs.
These utils provide autocompletion, event groups, and assertions to ensure proper usage.
The zone-less utils allow for selective disabling of zone patching for specific browser APIs, including requestAnimationFrame, cancelAnimationFrame, setInterval, clearInterval, setTimeout, clearTimeout.
The utility function `getZoneUnpatchedApi` can be used to get an unpatched version of any browser API.

#### [Coalescing:](https://www.rx-angular.io/docs/cdk/coalescing) Optimize Frequent Events

Coalescing sub-package provides set of utilities that implement a technique of merging multiple emissions, streams, or calls into one during a given time frame to improve the performance.
This helps to minimize the number of updates made to the user interface, improving the overall efficiency and speed of the application.
This technique is one of the key building blocks for optimizing change detection cycles in Angular, and is utilized to ensure that the application is functioning smoothly and effectively.

> #### [Coercing:](https://www.rx-angular.io/docs/cdk/coercing) Seamless Type Conversion
> 
> Our coercion package provides a set of utilities to perform type coercion (data type conversion) in a seamless and efficient manner.
> These utilities include factories, such as 'coerceObservable' and 'coerceDistinctObservable', and operators, such as 'coerceObservableWith' and 'coerceDistinctWith'.
> These tools help to ensure that data is in the desired format before it is used in the application, improving overall functionality and preventing unexpected errors.

Reviewer: ```suggestion
#### [Coercing:](https://www.rx-angular.io/docs/cdk/coercing) Seamless Type Conversion

Our coercion package provides a set of utilities to perform type coercion (data type conversion) in a seamless and efficient manner.
These utilities include factories, such as 'coerceObservable' and 'coerceDistinctObservable', and operators, such as 'coerceObservableWith' and 'coerceDistinctWith'.
These tools help to ensure that data is in the desired format before it is used in the application, improving overall functionality and preventing unexpected errors.

#### [Transformations](https://www.rx-angular.io/docs/cdk/api/transformation-helpers) Tree Shakable Entity Management

The transformation helpers provided by the `@rx-angular/cdk/transformations` are a complete set of utilities to manage entities. They are immutable by default and include transformation functions to handle insertions, updates, dictionary <-> array conversions & more to help you build store like state management systems.

```

---
title: Announcing the RxAngular CDK and Template 1.0 releases
description: The @rx-angular/cdk and @rx-angular/template packages are now stable and follow semantic versioning.
authors:
  - BioPhoton
  - edbzn
  - hoebbelsB
  - Karnaukhov-kh
  - LayZeeDK
tags: [cdk, release, template]
image: /img/logo.png
hide_table_of_contents: false
---

- [ ] Supported Angular versions
- [x] We follow Semantic Versioning
- [ ] Recent breaking changes/deprecations?
- [x] Introduction to each package
- [ ] Recent features in each package
- [x] Major features in each package

We are pleased to announce the stable release of RxAngular CDK and RxAngular Template 1.0. With this release, we enter semantic versioning which means no breaking changes in minor or patch version releases. This includes version requirements for our peer dependencies, Angular and RxJS.

> Thank you to all contributors and users who have helped us to get here. We are excited to see what the future holds for RxAngular.

Reviewer: ```suggestion
Thank you to all contributors and users who have helped us to get here. We are excited to see what the future holds for RxAngular.

## TOC

* [Template](#rx-angular-template)
* [CDK](#rx-angular-cdk)
* [Roadmap](#roadmap)
```

Coalescing sub-package provides set of utilities that implement a technique of merging multiple emissions, streams, or calls into one during a given time frame to improve the performance.
This helps to minimize the number of updates made to the user interface, improving the overall efficiency and speed of the application.
This technique is one of the key building blocks for optimizing change detection cycles in Angular, and is utilized to ensure that the application is functioning smoothly and effectively.

#### [Coercing:](https://www.rx-angular.io/docs/cdk/coercing) Seamless Type Conversion

Our coercion package provides a set of utilities to perform type coercion (data type conversion) in a seamless and efficient manner.
These utilities include factories, such as 'coerceObservable' and 'coerceDistinctObservable', and operators, such as 'coerceObservableWith' and 'coerceDistinctWith'.
These tools help to ensure that data is in the desired format before it is used in the application, improving overall functionality and preventing unexpected errors.

#### [Transformations](https://www.rx-angular.io/docs/cdk/api/transformation-helpers) Tree Shakable Entity Management

> The transformation helpers provided by the `@rx-angular/cdk/transformations` are a complete set of utilities to manage entities. They are immutable by default and include transformation functions to handle insertions, updates, dictionary <-> array conversions & more to help you build store like state management systems.

Reviewer: ```suggestion
The transformation helpers provided by the `@rx-angular/cdk/transformations` are a complete set of utilities to manage entities. They are immutable by default and include transformation functions to handle insertions, updates, dictionary <-> array conversions & more to help you build store like state management systems.

## Roadmap

We also want to give a brief overview about the future steps we want to take with the RxAngular libraries.

Right now, the RxAngular monorepo and all of its libraries are based on angular v12. This is because of backwards compatibility to pre-ivy projects.
While we like the idea of being backwards compatible, we have made the decision to move forward and upgrade the whole project to the latest angular version available. We want to make sure to build modern solutions based on the new APIs introduced with the latest changes to the angular framework itself.

Some things we are investigating right now:

* new provider functions (`withFeature`)
* easier creation of state (`inject`)
* signals integration into `@rx-angular/state`
* signals integration into `@rx-angular/template`
* concurrent mode with signals

That being said, we will release the next major versions very soon and we have decided to sync the `@rx-angular/*` versions with the angular versions, giving us the ability to provide two breaking changes a year.

| RxAngular  | Angular version  |
|------------|------------------|
| `1.0.0`    | `>= 12`          |
| `2.0.0`    | `>= 13`          |
| `14.0.0`   | `^14`            |
| `15.0.0`   | `^15`            |
| `16.x`     | `^16`            |

```

Our coercion package provides a set of utilities to perform type coercion (data type conversion) in a seamless and efficient manner.
These utilities include factories, such as 'coerceObservable' and 'coerceDistinctObservable', and operators, such as 'coerceObservableWith' and 'coerceDistinctWith'.
These tools help to ensure that data is in the desired format before it is used in the application, improving overall functionality and preventing unexpected errors.

#### [Transformations](https://www.rx-angular.io/docs/cdk/api/transformation-helpers) Tree Shakable Entity Management

The transformation helpers provided by the `@rx-angular/cdk/transformations` are a complete set of utilities to manage entities. They are immutable by default and include transformation functions to handle insertions, updates, dictionary <-> array conversions & more to help you build store like state management systems.

## Roadmap

We also want to give a brief overview about the future steps we want to take with the RxAngular libraries.

Right now, the RxAngular monorepo and all of its libraries are based on angular v12. This is because of backwards compatibility to pre-ivy projects.

> While we like the idea of being backwards compatible, we have made the decision to move forward and upgrade the whole project to the latest angular version available. We want to make sure to build modern solutions based on the new APIs introduced with the latest changes to the angular framework itself.

Reviewer 2: ```suggestion
While we like the idea of being backwards compatible, we have made the decision to move forward and upgrade the whole project to the latest Angular version available. We want to make sure to build modern solutions based on the new APIs introduced with the latest changes to the Angular framework itself.
```

Some things we are investigating right now:

* new provider functions (`withFeature`)
* easier creation of state (`inject`)
* signals integration into `@rx-angular/state`
* signals integration into `@rx-angular/template`
* concurrent mode with signals

> That being said, we will release the next major versions very soon and we have decided to sync the `@rx-angular/*` versions with the angular versions, giving us the ability to provide two breaking changes a year.

Reviewer 2: ```suggestion
That being said, we will release the next major versions very soon and we have decided to sync the `@rx-angular/*` versions with the Angular versions, giving us the ability to provide two breaking changes a year.
```
